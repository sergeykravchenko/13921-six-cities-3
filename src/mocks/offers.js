export default [
  {
    "id": 1,
    "zoom": 15,
    "name": `Canal View Prinsengracht`,
    "picture": `img/room.jpg`,
    "price": 120,
    "priceText": `night`,
    "rating": 4.8,
    "features": {
      "type": `Private room`,
      "bedrooms": 3,
      "maxGuests": 3,
    },
    "isPremium": true,
    "isInBookmark": true,
    "houseHolds": [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    "images": [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/room.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-03.jpg`,
    ],
    "host": {
      "id": 2,
      "name": `Max`,
      "img": `img/avatar-max.jpg`,
      "isPro": true,
    },
    "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    "coords": [52.3909553943508, 4.85309666406198],
    "reviews": [
      {
        "id": 1,
        "author": `Max`,
        "avatar": `img/avatar-max.jpg`,
        "rating": 4,
        "text": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                 The building is green and from 18th century.`,
        "date": new Date(`2019-04-10`),
      }, {
        "id": 2,
        "author": `Angelina`,
        "avatar": `img/avatar-angelina.jpg`,
        "rating": 5,
        "text": `Picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
        "date": new Date(`2019-04-10`),
      }
    ],
    "city": {
      "name": `Amsterdam`,
      "coords": [52.38333, 4.9],
    },
  }, {
    "id": 2,
    "zoom": 13,
    "name": `Nice, cozy, warm big bed apartment`,
    "picture": `img/apartment-02.jpg`,
    "price": 80,
    "priceText": `night`,
    "rating": 5,
    "features": {
      "type": `Apartment`,
      "bedrooms": 3,
      "maxGuests": 3,
    },
    "houseHolds": [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
    ],
    "images": [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/room.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-03.jpg`,
    ],
    "host": {
      "id": 3,
      "name": `Angelina`,
      "img": `img/avatar-angelina.jpg`,
      "isPro": false,
    },
    "description": `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    "coords": [52.369553943508, 4.85309666406198],
    "reviews": [
      {
        "id": 1,
        "author": `Angelina`,
        "avatar": `img/avatar-angelina.jpg`,
        "rating": 4.5,
        "text": `Picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
        "date": new Date(`2019-04-8`),
      }
    ],
    "city": {
      "name": `Amsterdam`,
      "coords": [52.38333, 4.9],
    },
  }, {
    "id": 3,
    "zoom": 11,
    "name": `Wood and stone place`,
    "picture": `img/room.jpg`,
    "price": 132,
    "priceText": `night`,
    "rating": 5,
    "features": {
      "type": `Private room`,
      "bedrooms": 4,
      "maxGuests": 3,
    },
    "isPremium": true,
    "houseHolds": [
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
    ],
    "images": [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/room.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-03.jpg`,
    ],
    "host": {
      "id": 4,
      "name": `Max`,
      "img": `img/avatar-max.jpg`,
      "isPro": false,
    },
    "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    "coords": [52.3909553943508, 4.929309666406198],
    "reviews": [
      {
        "id": 1,
        "author": `Max`,
        "avatar": `img/avatar-max.jpg`,
        "rating": 4,
        "text": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                 The building is green and from 18th century.`,
        "date": new Date(`2019-04-24`),
      }
    ],
    "city": {
      "name": `Cologne`,
      "coords": [52.38343, 4.9],
    }
  }, {
    "id": 4,
    "zoom": 18,
    "name": `Beautiful &amp; luxurious apartment at great location`,
    "picture": `img/apartment-01.jpg`,
    "price": 180,
    "priceText": `night`,
    "rating": 4,
    "features": {
      "type": `Apartment`,
      "bedrooms": 5,
      "maxGuests": 3,
    },
    "houseHolds": [
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
    ],
    "images": [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/room.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-03.jpg`,
    ],
    "host": {
      "id": 1,
      "name": `Angelina`,
      "img": `img/avatar-angelina.jpg`,
      "isPro": true,
    },
    "description": `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    "coords": [52.3809553943508, 4.939309666406198],
    "city": {
      "name": `Hamburg`,
      "coords": [52.38013, 4.9],
    },
  },
];
