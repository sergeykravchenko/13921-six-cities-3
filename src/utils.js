export const SortType = {
  POPULAR: `Popular`,
  PRICE_TO_HIGH: `Price: low to high`,
  PRICE_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

