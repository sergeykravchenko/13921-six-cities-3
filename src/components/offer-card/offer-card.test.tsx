import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {Router} from "react-router-dom";
import history from "../../history.js";
import NameSpace from "../../reducer/name-space";
import OfferCard from './offer-card';
import offers from '../../mocks/offers';
import {noop} from "../../utils";

const mockStore = configureStore([]);

const offer = {
  "id": 1,
  "name": `Canal View Prinsengracht`,
  "picture": `img/room.jpg`,
  "price": 120,
  "priceText": `night`,
  "rating": 4,
  "zoom": 16,
  "features": {
    "type": `Private room`,
    "bedrooms": 8,
    "maxGuests": 7,
  },
  "isPremium": true,
  "isInBookmark": true,
  "houseHolds": [
    `Wi-Fi`,
    `Washing machine`,
  ],
  "images": [
    `img/apartment-01.jpg`,
    `img/apartment-02.jpg`,
  ],
  "host": {
    "id": 2,
    "name": `Max`,
    "img": `img/avatar-max.jpg`,
    "isPro": true,
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  "coords": [23, 12],
};

const isAuthenticated = false;
const activeMarker = 3;

const activeCity = {
  "name": `Hamburg`,
  "coords": [52.38013, 4.9],
};

const match = {
  params: {
    id: 2,
  }
};

const isFetching = false;

it(`Offer-card renders correctly`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      allOffers: [],
      neighbors: [],
      comments: [],
    },
    [NameSpace.STATE]: {
      requestStatus: ``,
      activeOffer: offer,
    },
    [NameSpace.USER]: {
      authorizationStatus: `AUTH`,
      user: {
        "avatar_url": `img/1.png`,
        "email": `Oliver.conner@gmail.com`,
        "id": 1,
        "is_pro": false,
        "name": `Oliver.conner`,
      },
    },
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <Router
            history={history}
          >
            <OfferCard
              isFetching={isFetching}
              offers={offers}
              isAuthenticated={isAuthenticated}
              activeOffer={offer}
              neighbors={offers.slice(0, 2)}
              activeMarker={activeMarker}
              activeCity={activeCity}
              onLoadData={noop}
              match={match}
            />,
          </Router>
        </Provider>, {
          createNodeMock: () => {
            return document.createElement(`div`);
          }})
    .toJSON();
  expect(tree).toMatchSnapshot();
});
