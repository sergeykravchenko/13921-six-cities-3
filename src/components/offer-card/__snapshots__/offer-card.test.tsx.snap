// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offer-card renders correctly 1`] = `
Array [
  <div
    className="page"
  >
    <header
      className="header"
    >
      <div
        className="container"
      >
        <div
          className="header__wrapper"
        >
          <div
            className="header__left"
          >
            <a
              className="header__logo-link"
              href="/"
              onClick={[Function]}
            >
              <img
                alt="6 cities logo"
                className="header__logo"
                height="41"
                src="/img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            className="header__nav"
          >
            <ul
              className="header__nav-list"
            >
              <li
                className="header__nav-item user"
              >
                <a
                  className="header__nav-link header__nav-link--profile"
                  href="/favorites"
                  onClick={[Function]}
                >
                  <div
                    className="header__avatar-wrapper user__avatar-wrapper"
                    style={
                      Object {
                        "backgroundImage": "url(https://htmlacademy-react-3.appspot.com/six-citiesimg/1.png)",
                        "borderRadius": "50%",
                      }
                    }
                  />
                  <span
                    className="header__login"
                  >
                    Oliver.conner@gmail.com
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      className="page__main page__main--property"
      id="2"
    >
      <section
        className="property"
      >
        <div
          className="property__gallery-container container"
        >
          <div
            className="property__gallery"
          >
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="img/apartment-01.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="img/apartment-02.jpg"
              />
            </div>
          </div>
        </div>
        <div
          className="property__container container"
        >
          <div
            className="property__wrapper"
          >
            <div
              className="property__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="property__name-wrapper"
            >
              <h1
                className="property__name"
              >
                Canal View Prinsengracht
              </h1>
              <button
                className="property__bookmark-button
                  property__bookmark-button--active button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="33"
                  width="31"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="property__rating rating"
            >
              <div
                className="property__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "80%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
              <span
                className="property__rating-value rating__value"
              >
                4
              </span>
            </div>
            <ul
              className="property__features"
            >
              <li
                className="property__feature property__feature--entire"
              >
                Private room
              </li>
              <li
                className="property__feature property__feature--bedrooms"
              >
                8
                 Bedrooms
              </li>
              <li
                className="property__feature property__feature--adults"
              >
                Max 
                7
                 adults
              </li>
            </ul>
            <div
              className="property__price"
            >
              <b
                className="property__price-value"
              >
                €
                120
              </b>
              <span
                className="property__price-text"
              >
                 
                night
              </span>
            </div>
            <div
              className="property__inside"
            >
              <h2
                className="property__inside-title"
              >
                What's inside
              </h2>
              <ul
                className="property__inside-list"
              >
                <li
                  className="property__inside-item"
                >
                  Wi-Fi
                </li>
                <li
                  className="property__inside-item"
                >
                  Washing machine
                </li>
              </ul>
            </div>
            <div
              className="property__host"
            >
              <h2
                className="property__host-title"
              >
                Meet the host
              </h2>
              <div
                className="property__host-user user"
              >
                <div
                  className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
                >
                  <img
                    alt="Host avatar"
                    className="property__avatar user__avatar"
                    height="74"
                    src="/undefined"
                    width="74"
                  />
                </div>
                <span
                  className="property__user-name"
                >
                  Max
                </span>
              </div>
              <div
                className="property__description"
              >
                <p
                  className="property__text"
                >
                  A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
                </p>
              </div>
            </div>
            <section
              className="property__reviews reviews"
            >
              <h2
                className="reviews__title"
              >
                Reviews ·
                <span
                  className="reviews__amount"
                >
                  0
                </span>
              </h2>
              <ul
                className="reviews__list"
              />
              <form
                action="#"
                className="reviews__form form"
                method="post"
                onSubmit={[Function]}
              >
                
                <label
                  className="reviews__label form__label"
                  htmlFor="review"
                >
                  Your review
                </label>
                <div
                  className="reviews__rating-form form__rating"
                >
                  <input
                    checked={false}
                    className="form__rating-input visually-hidden"
                    disabled=""
                    id="5-stars"
                    name="rating"
                    onChange={[Function]}
                    type="radio"
                    value={5}
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="5-stars"
                    title="perfect"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    checked={false}
                    className="form__rating-input visually-hidden"
                    disabled=""
                    id="4-stars"
                    name="rating"
                    onChange={[Function]}
                    type="radio"
                    value={4}
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="4-stars"
                    title="perfect"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    checked={false}
                    className="form__rating-input visually-hidden"
                    disabled=""
                    id="3-stars"
                    name="rating"
                    onChange={[Function]}
                    type="radio"
                    value={3}
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="3-stars"
                    title="perfect"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    checked={false}
                    className="form__rating-input visually-hidden"
                    disabled=""
                    id="2-stars"
                    name="rating"
                    onChange={[Function]}
                    type="radio"
                    value={2}
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="2-stars"
                    title="perfect"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    checked={false}
                    className="form__rating-input visually-hidden"
                    disabled=""
                    id="1-stars"
                    name="rating"
                    onChange={[Function]}
                    type="radio"
                    value={1}
                  />
                  <label
                    className="reviews__rating-label form__rating-label"
                    htmlFor="1-stars"
                    title="perfect"
                  >
                    <svg
                      className="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlinkHref="#icon-star"
                      />
                    </svg>
                  </label>
                </div>
                <textarea
                  className="reviews__textarea form__textarea"
                  disabled=""
                  id="review"
                  name="review"
                  onChange={[Function]}
                  placeholder="Tell how was your stay, what you like and what can be improved"
                  value=""
                />
                <div
                  className="reviews__button-wrapper"
                >
                  <p
                    className="reviews__help"
                  >
                    To submit review please make sure to set 
                    <span
                      className="reviews__star"
                    >
                      rating
                    </span>
                     and describe your stay with at least 
                    <b
                      className="reviews__text-amount"
                    >
                      50 characters
                    </b>
                    .
                  </p>
                  <button
                    className="reviews__submit form__submit button"
                    disabled={true}
                    type="submit"
                  >
                    Submit
                  </button>
                </div>
              </form>
            </section>
          </div>
        </div>
        <section
          className="property__map map"
        />
      </section>
      <div
        className="container"
      >
        <section
          className="near-places places"
        >
          <h2
            className="near-places__title"
          >
            Other places in the neighbourhood
          </h2>
          <div
            className="near-places__list places__list"
          />
        </section>
      </div>
    </main>
  </div>,
  ",",
]
`;
